src/components/user/user.service.ts

```diff
@@ -174,6 +177,16 @@ export class UserService {
     ];
   };
 
+  roleProperties = (roles?: Role[]) => {
+    let roleProperties: any[] = [];
+    roles?.forEach(
+      (role) =>
+        (roleProperties = roleProperties.concat(this.property('roles', role)))
+    );
+
+    return roleProperties;
+  };
```

Comment:
Usually `.forEach` can be replaced by `.map` since forEach returns void.
I've never had a need to use `.forEach`.

```suggestion
  private roleProperties(roles?: Role[]) {
    return (roles || []).flatMap(role => this.property('roles', role));
  }
```